packages/plugin-session-replay-browser/src/session-replay.ts

```diff
@@ -76,39 +76,46 @@ class SessionReplay implements SessionReplayEnrichmentPlugin {
       };
     }
 
-    const GlobalScope = getGlobalScope();
-    if (GlobalScope && GlobalScope.window) {
-      GlobalScope.window.addEventListener('blur', () => {
-        this.stopRecordingEvents && this.stopRecordingEvents();
-        this.stopRecordingEvents = null;
-      });
-      GlobalScope.window.addEventListener('focus', () => {
-        void this.initialize();
-      });
+    const globalScope = getGlobalScope();
+    if (globalScope) {
+      globalScope.addEventListener('blur', this.blurListener);
+      globalScope.addEventListener('focus', this.focusListener);
     }
 
-    if (GlobalScope && GlobalScope.document && GlobalScope.document.hasFocus()) {
+    if (globalScope && globalScope.document && globalScope.document.hasFocus()) {
       await this.initialize(true);
     }
   }
 
+  blurListener = () => {
+    this.stopRecordingAndSendEvents();
+  };
+  focusListener = () => {
+    void this.initialize();
+  };
+
+  stopRecordingAndSendEvents(sessionId?: number) {
```

Comment:
Consolidating logic here, and actually discovered a bug doing so - any time we call `stopRecordingEvents`, we have to set it to null afterwards to signal that recording can begin again